src/test/java/de/tum/in/www1/artemis/ModelingSubmissionIntegrationTest.java

```diff
@@ -387,19 +492,19 @@ public void getNextOptimalModelSubmission_lockLimitReached() throws Exception {
     }
 
     @Test
-    @WithMockUser(value = "student1")
-    public void getModelSubmissionForModelingEditor() throws Exception {
-        ModelingSubmission submission = ModelFactory.generateModelingSubmission(validModel, true);
-        submission = database.addModelingSubmissionWithFinishedResultAndAssessor(classExercise, submission, "student1", "tutor1");
+    @WithMockUser(value = "tutor1", roles = "TA")
+    public void deleteNextOptimalModelSubmission() throws Exception {
+        request.delete("/api/exercises/" + classExercise.getId() + "/optimal-model-submissions", HttpStatus.NO_CONTENT);
```

Comment:
I think you should invoke `@GetMapping(value = "/exercises/{exerciseId}/modeling-submission-without-assessment")` once before, so that there are optimal submissions calculated on the server.

Then please also check that the list of optimal models is empty. You might need to add a public getter to obtain this information, it is stored in CompassService (can be injected) -> CompassCalculationEngine -> ModelSelector -> modelsWaitingForAssessment

Reply:
@krusche Thank you for your feedback! I have adjusted the test and added the mentioned REST calls :)